src/main/java/com/cloudbees/jenkins/plugins/mtslavescloud/MansionCloud.java

```diff
@@ -289,9 +293,68 @@ public Credential createAccessToken(URL broker) throws AbortException, OauthClie
         }
         label = Jenkins.getInstance().getLabel(st.getLabel()+" "+box.size+compat);
 
-        
+
         final Queue<PlannedMansionSlave> queue = new ArrayBlockingQueue<PlannedMansionSlave>(excessWorkload);
         List<PlannedNode> r = new ArrayList<PlannedNode>();
+
+        /**
+         * The current approach - fire up as much slaves as we can - is pretty broken.
+         * Does not respect the mansion's resources and actually causes a lot of timeouts,
+         * and leaves a lot of lxcs in a pretty bad state.
+         *
+         * A much better approach is to limit the number of provisioned slaves per M min.
+         * Initially fire up N slaves. After S seconds (60-120-240?) if the build queue is still there,
+         * fire up another (N-1) slaves. Increase the number of the slaves gradually and not in bursts.
+         *
+         * The customers will appreciate it, because they don't want to see a lot of failed provisioning requests,
+         * but stable, running builds which can be achieved by the gradually started slaves.
+         *
+         * This change not also reduces the load on the mansions, but on the masters as well.
+         *
+         */
+        long currentEpoch = System.currentTimeMillis()/1000;
+
+        int currentNumberOfSlaves = 0;
+        for (MansionSlave n : Util.filter(Jenkins.getInstance().getNodes(), MansionSlave.class)) {
+            currentNumberOfSlaves++;
+
+        }
+        LOGGER.log(Level.INFO, "We have {0} slaves, excessWorkload = {1}", new Object[]{currentNumberOfSlaves, excessWorkload});
+        // If we don't have slaves at all we need to launch immediately
+        if ( currentNumberOfSlaves == 0 ) {
+            //Reduce the number of the initial slaves only when it wants to fire up too many at the same time,
+            if ( excessWorkload > INITIAL_SLAVES_TO_START ) {
```

Comment:
Does Mansion plugin set the strict limit on executor number for each slave. This logic is designed for single-executor slaves. If there are ore slaves, the logic may cause the overprovisioning